lib/matplotlib/artist.py

```diff
@@ -43,12 +53,32 @@ def draw_wrapper(artist, renderer, *args, **kwargs):
             if artist.get_agg_filter() is not None:
                 renderer.stop_filter(artist.get_agg_filter())
             if artist.get_rasterized():
+                renderer._raster_depth -= 1
+            if (renderer._rasterizing and artist.figure and
+                    artist.figure.suppressComposite):
+                # restart rasterizing to prevent merging
                 renderer.stop_rasterizing()
+                renderer.start_rasterizing()
 
     draw_wrapper._supports_rasterization = True
     return draw_wrapper
 
 
+def finalize_rasterization(draw):
```

Comment:
Moving this out seems reasonable, I think this should be private (I doubt you want to let any third-party artist use that?)

Reply:
Makes sense. I'll do that now.